src/sentry_plugins/github/endpoints/webhook.py

```diff
@@ -273,9 +273,120 @@ def __call__(self, event, organization=None):
             self._handle(event, org, is_apps)
 
 
+class PullRequestEventWebhook(Webhook):
+    # https://developer.github.com/v3/activity/events/types/#pullrequestevent
+    def __call__(self, event, organization):
+        action = event['action']
+        is_apps = 'installation' in event
+
+        try:
+            repo = Repository.objects.get(
+                organization_id=organization.id,
+                provider='github_apps' if is_apps else 'github',
+                external_id=six.text_type(event['repository']['id']),
+            )
+
+        except Repository.DoesNotExist:
+            raise Http404()
+
+        # We need to track GitHub's "full_name" which is the repository slug.
+        # This is needed to access the API since `external_id` isn't sufficient.
+        if repo.config.get('name') != event['repository']['full_name']:
+            repo.config['name'] = event['repository']['full_name']
+            repo.save()
+
+        if action == 'opened':
+            self._handle_created(event, organization, repo, is_apps)
+        if action == 'edited':
+            self._handle_updated(event, organization, repo)
+        if action == 'synchronize':
+            self._handle_sync(event, organization, repo)
+
+    def _handle_created(self, event, organization, repo, is_apps):
+        """PR was created"""
+
+        client = GitHubClient()
+
+        pull_request = event['pull_request']
+        number = pull_request['number']
+        title = pull_request['title']
+        body = pull_request['body']
+        user = pull_request['user']
+
+        # get author (TODO: use more complicated logic from the commit webhook)
+        author = author = CommitAuthor.objects.get_or_create(
+            organization_id=organization.id,
+            external_id=user['id'],
+            defaults={
+                'name': user['login']
+            }
+        )[0]
+
+        pr = ChangeRequest.objects.create(
+            repository_id=repo.id,
+            organization_id=organization.id,
+            key=number,
+            title=title,
+            message=body,
+            author=author
+        )
+
+        commit_url = pull_request['commits_url']
+        commits = client.request('GET', commit_url)
+
+        pr.save()
+        pr.commits.clear()
+        commits = Commit.objects.filter(
+            key__in=[c['sha'] for c in commits]
+        )
+        pr.commits.add(*commits)
+
+    def _handle_updated(self, event, organization, repo):
+        """PR title or description was edited"""
+
+        pull_request = event['pull_request']
+        number = pull_request['number']
+        title = pull_request['title']
+        body = pull_request['body']
+
+        pr = ChangeRequest.objects.get(
```

Comment:
you should handle a `ChangeRequest.DoesNotExist` error

Reply:
you also could probably combine this and the update to a single query by doing:
```python
ChangeRequest.objects.filter(
    repository_id=repo.id,
    key=number,
).update(
    title=title,
    message=body,
)

```

Reply:
did you mean get or filter here? those are unique together keys i think